nautobot/extras/forms/mixins.py

```diff
@@ -0,0 +1,587 @@
+                        source_type=self.obj_type,
+                        source_id=self.instance.pk,
+                        destination_type=self.obj_type,  # since this is a symmetric relationship this is OK
+                        destination_id=peer_id,
+                    )
+
+                association.clean()
+                association.save()
+
+    def save(self, commit=True):
+
+        obj = super().save(commit)
+        if commit:
+            self._save_relationships()
+
+        return obj
+
+
+class RelationshipModelFilterFormMixin(forms.Form):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.relationships = []
+        self.obj_type = ContentType.objects.get_for_model(self.model)
+        self._append_relationships()
+
+    def _append_relationships(self):
+        """
+        Append form fields for all Relationships assigned to this model.
+        """
+        source_relationships = Relationship.objects.filter(source_type=self.obj_type, source_hidden=False)
+        self._append_relationships_side(source_relationships, RelationshipSideChoices.SIDE_SOURCE)
+
+        dest_relationships = Relationship.objects.filter(destination_type=self.obj_type, destination_hidden=False)
+        self._append_relationships_side(dest_relationships, RelationshipSideChoices.SIDE_DESTINATION)
+
+    def _append_relationships_side(self, relationships, initial_side):
+        """
+        Helper method to _append_relationships, for processing one "side" of the relationships for this model.
+        """
+        for relationship in relationships:
+            if relationship.symmetric:
+                side = RelationshipSideChoices.SIDE_PEER
+            else:
+                side = initial_side
+            peer_side = RelationshipSideChoices.OPPOSITE[side]
+
+            # If this model is on the "source" side of the relationship, then the field will be named
+            # "cr_<relationship-slug>__destination" since it's used to pick the destination object(s).
+            # If we're on the "destination" side, the field will be "cr_<relationship-slug>__source".
+            # For a symmetric relationship, both sides are "peer", so the field will be "cr_<relationship-slug>__peer"
+            field_name = f"cr_{relationship.slug}__{peer_side}"
+
+            if field_name in self.relationships:
+                # This is a symmetric relationship that we already processed from the opposing "initial_side".
+                # No need to process it a second time!
+                continue
+
+            self.fields[field_name] = relationship.to_form_field(side=side)
+            if relationship.source_label and relationship.destination_label:
+                self.fields[
+                    field_name
+                ].label = f"{relationship.source_label} to {relationship.destination_label}__{peer_side}"
+            else:
+                self.fields[field_name].label = field_name
```

Comment:
This should just be something like `self.fields[field_name].label = relationship.get_label(side)` -- which actually `relationship.to_form_field` already provides for you, so this shouldn't be needed at all.